tech_reports/MetalProfiler/metal-profiler.md

```diff
@@ -0,0 +1,331 @@
+
+### 5. (Only if did 1.) View .tracy contents 
+You can also view the contents of the .tracy file as a csv file using tracy-csvexport. This will dump the results in csv format which you can pip into a file and view the results. Optionally, you can also save the .tracy file via the GUI itself and then feed it into the tracy-csvexport tool.
+```
+./tracy-csvexport hello.tracy
+```
+
+## Tracy Example
+The following section will provide an example of how to use Tracy in a sample app, step by step.
+
+### 1. Setup project directory structure
+```
+- project/
+  - third_party/
+    - tracy/
+  - include/
+    - hellolib.hpp
+  - CMakeLists.txt
+  - main.cpp
+  - hellolib.cpp 
+```
+
+### 2. Fill in contents of each file found below
+// third_party/tracy
+```
+mkdir third_party
+cd third_party
+git clone https://github.com/tenstorrent-metal/tracy/tree/71d4c8d378b52af7da7012b9b595a61e9304f0bb
+cd tracy
+```
+
+// hellolib.hpp
+```
+#include "tracy/Tracy.hpp"
+#include <string>
+
+int add(int a, int b);
+int subtract(int a, int b);
+int multiply(int a, int b);
+int divide(int a, int b);
+```
+
+// CMakeLists.txt
+```
+cmake_minimum_required(VERSION 3.10)
+
+project(TracyTest LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DTRACY_ENABLE=ON")
+
+add_library(hellolib SHARED hellolib.cpp)
+
+target_include_directories(hellolib
+    PUBLIC 
+        ${CMAKE_SOURCE_DIR}/include
+        ${CMAKE_SOURCE_DIR}/third_party/tracy/public
+)
+
+add_executable(runner main.cpp)
+
+add_subdirectory(third_party/tracy)
+
+# Link the executable against the shared library
+target_link_libraries(runner PUBLIC hellolib Tracy::TracyClient)
+```
+
+// main.cpp
+```
+#include "tracy/Tracy.hpp"
+#include <iostream>
+#include "hellolib.hpp"
+#include <unistd.h> // for sleep()
+
+int main() {
+    sleep(5); // need to add short sleep call so that tracy client can establish connection
+    FrameMarkNamed("main");
+    TracyMessageL("Hello");
+    ZoneScoped;
+    int c = 0;
+    std::cout << "Hello, World = " << c << std::endl;
+    c = add(5, 5);
+    c = subtract(5, 5);
+    c = multiply(5, 5);
+    c = divide(5, 5);
+    return 0;
+}
+```
+
+// hellolib.cpp
+```
+#include "tracy/Tracy.hpp"
+#include "hellolib.hpp"
+#include <iostream>
+#include <string>
+
+int add(int a, int b) {
+  TracyMessageL("add");
+  ZoneScoped;
+  int c = a + b;
+  return c;
+}
+
+int subtract(int a, int b) {
+  TracyMessageL("subtract");
+  ZoneScoped;
+  return a - b;
+}
+
+int multiply(int a, int b) {
+  TracyMessageL("multiply");
+  ZoneScoped;
+  return a * b;
+}
+
+int divide(int a, int b) {
+  TracyMessageL("divide");
+  ZoneScoped;
+  return a / b;
+}
+```
+
+### 3. Build Project
+This will build the executable and binarines in `build/` folder.
+```
+mkdir -p build 
+cd build/
+cmake -G Ninja -DTRACY_ENABLE=ON .. 
+ninja 
+```
+
+### 4. Build tracy-capture
```

Comment:
Unused headers

Reply:
sorry what do you mean by unused headers?